Dockerfile.ci

```diff
@@ -0,0 +1,29 @@
+# This is the image pushed to Dockerhub, containing the built and tested Locust package
+
+# Stage 1: Install Locust package
+FROM python:3.11-slim AS base
+
+FROM base AS builder
+RUN apt-get update && apt-get install -y git 
```

Comment:
What's the reason for installing git here? Where is it used?

Reply:
No idea! This is cruft from the historical docker image, I assumed the building of binaries required it

Reply:
It was needed so that we can determine the locust version name (it was used by setuptools_scm), e.g. for use with `locust --version`. Does it still work with the new way of building the docker image?

Reply:
I believe the Dynamic Versioning plugin for Poetry comes with whatever it needs 🤔 
--- but I'll check on the docker image, that a produced version is correct

On the CI, I did make a cut-down docker image which doesn't do any building at all, just takes the already-known-good built assets - the version of this is already resolved in the package (and can be seen in the test-docker-image step)
![image](https://github.com/user-attachments/assets/4d626fff-46ce-4829-9ff7-ce7789d2f9fc)

On the "normal" image, which users can still build end-to-end, I can check that the output of `locust --version` is correct inside the docker image 🤔

Reply:
It turns out the git binary is a necessity for resolving the version (understandably) even using `dunamai` etc. so I guess it gets left in, wanted to have the images as close to each other as possible.

Reply:
This does work with the local build as well, seems to get the correct version
![image](https://github.com/user-attachments/assets/864ecce0-f63c-41cb-a729-540b84c2eac2)

Reply:
Addressed in https://github.com/locustio/locust/pull/2873/commits/7d7bfc673d7b2676d3491d8f8218510d8d1bfa49

(had to allow `dist` in the docker context for the CI, so the local one has to remove it (or declare a wheel file name specifically)